xbmc/commons/Buffer.h

```diff
@@ -163,6 +172,14 @@ namespace XbmcCommons
       return *this;
     }
 
+    /**
+    * Copy data into and return std::vector<char>
+    */
+    inline operator std::vector<char>() const
+    {
+      return std::vector<char>(buffer, buffer + mcapacity);
+    }
```

Comment:
likewise this seem superfluous. Does it make sense for Buffer to understand std:vector<char> but not other std:vector<...>. If anything std::vector<unsigned char> makes a bit more sense since `unsigned char` is synonymous with a byte.